src/mixins/filing-template-mixin.ts

```diff
@@ -129,7 +130,8 @@ export default class FilingTemplateMixin extends Mixins(DateMixin, EnumMixin) {
         contactPoint: this.getContactPoint,
         nameRequest: this.getNameRequest,
         offices: this.getOfficeAddresses,
-        type: this.getCorrectionErrorType
+        type: this.getCorrectionErrorType,
+        ...(this.isEntityTypeFirm && { startDate: this.getCorrectionStartDate })
```

Comment:
I think this isn't needed since you have the code on lines 191-193 already.

Reply:
Is this a cleaner implementation that I could use instead of the code on lines 191-193?

Reply:
It's OK, but ...

First, does this work? It looks like you are destructuring a `()` block...?

Second, you may need to add a condition for whether Start Date has changed (so you don't save it if not).

Third, I think it is actually cleaner to conditionally add something (as below) instead of conditionally adding a key or undefined.

Reply:
Yes it uses the spread operator, so that the so the object `{ startDate: value}` is returned only if the condition is true and then it is spread into the object.
Okay I will change this to conditionally add it as it is done below.